packages/sdk/src/protocols/filter.ts

```diff
@@ -143,16 +147,39 @@ export class SubscriptionManager implements ISubscriptionSDK {
     return finalResult;
   }
 
-  async ping(): Promise<SDKProtocolResult> {
-    const promises = this.peers.map(async (peer) => this.protocol.ping(peer));
+  async ping(peerId?: PeerId): Promise<SDKProtocolResult> {
+    if (peerId) {
+      const peer = this.getPeers().find((p) => p.id.equals(peerId));
```

Comment:
curious to know why we use `p.id.equals`

Reply:
we rarely use it's API so I was confused, why did you prefer it over `=`? just curious

Reply:
I think `peerId` here has `toString` defined which is compatible with `===` but I don't care if `.equals` is used